src/Xamarin.Android.Build.Tasks/Tasks/ResolveLibraryProjectImports.cs

```diff
@@ -366,7 +366,7 @@ public override bool Execute ()
 
 				bool updated = false;
 				string aarHash = MonoAndroidHelper.HashFile (aarFile.ItemSpec);
-				string stamp = Path.Combine (outdir, Path.GetFileNameWithoutExtension (aarFile.ItemSpec) + ".stamp");
+				string stamp = Path.Combine (outdir, aarIdentityName + ".stamp");
```

Comment:
So what will happen is a user has `__AndroidLibraryProjects__ ` AND a `AndroidAarLibrary`... Will they both try to end up in the same directory?

Reply:
Seems they all end up in a <number> directory...

Reply:
Is that what we want? I'm curious what will happen if they contain similar files.. do we overwrite or does it error.

Reply:
Maybe I wasn't clear in the commit message, in how it was broken.

We return these as the stamp files: https://github.com/xamarin/xamarin-android/blob/3c06da0228032cda11f256f7fb34723cf2774f68/src/Xamarin.Android.Build.Tasks/Tasks/ResolveLibraryProjectImports.cs#L110-L112

But we were using the filename of the AAR as the stamp:
https://github.com/xamarin/xamarin-android/blob/3c06da0228032cda11f256f7fb34723cf2774f68/src/Xamarin.Android.Build.Tasks/Tasks/ResolveLibraryProjectImports.cs#L369

Which doesn't match up what we do for .NET assemblies:
https://github.com/xamarin/xamarin-android/blob/3c06da0228032cda11f256f7fb34723cf2774f68/src/Xamarin.Android.Build.Tasks/Tasks/ResolveLibraryProjectImports.cs#L206

So the new test, was finding files here that just didn't exist at all:
```csharp
foreach (var s in actual.ResolvedResourceDirectoryStamps) {
	FileAssert.Exists (s.ItemSpec);
}
```
And these trigger targets to always re-run if you are using an AAR file in your app...

Reply:
@jonathanpeppers I know, one of the reasons why I think I used the assembly name for the aar was so that the resources didn't get mixed up between aar and assembly... not sure that was valid or not. I just wanted to check what would happen if a Nuget Package had both Aar AND the normal embedded libraries, which is something a user can do.

Reply:
I re-read my comment and I think I sent the wrong message. It looks to be working properly. Good stuff!

Reply:
It was worth checking, but if I look at `obj\Debug\lp\map.cache` from this test:
```
FormsViewGroup
Xamarin.Android.Arch.Core.Common
Xamarin.Android.Arch.Core.Runtime
Xamarin.Android.Arch.Lifecycle.Common
Xamarin.Android.Arch.Lifecycle.LiveData.Core
Xamarin.Android.Arch.Lifecycle.LiveData
Xamarin.Android.Arch.Lifecycle.Runtime
Xamarin.Android.Arch.Lifecycle.ViewModel
Xamarin.Android.Support.Animated.Vector.Drawable
Xamarin.Android.Support.Annotations
Xamarin.Android.Support.AsyncLayoutInflater
Xamarin.Android.Support.Collections
Xamarin.Android.Support.Compat
Xamarin.Android.Support.CoordinaterLayout
Xamarin.Android.Support.Core.UI
Xamarin.Android.Support.Core.Utils
Xamarin.Android.Support.CursorAdapter
Xamarin.Android.Support.CustomTabs
Xamarin.Android.Support.CustomView
Xamarin.Android.Support.Design
Xamarin.Android.Support.DocumentFile
Xamarin.Android.Support.DrawerLayout
Xamarin.Android.Support.Fragment
Xamarin.Android.Support.Interpolator
Xamarin.Android.Support.Loader
Xamarin.Android.Support.LocalBroadcastManager
Xamarin.Android.Support.Media.Compat
Xamarin.Android.Support.Print
Xamarin.Android.Support.SlidingPaneLayout
Xamarin.Android.Support.SwipeRefreshLayout
Xamarin.Android.Support.Transition
Xamarin.Android.Support.v4
Xamarin.Android.Support.v7.AppCompat
Xamarin.Android.Support.v7.CardView
Xamarin.Android.Support.v7.MediaRouter
Xamarin.Android.Support.v7.Palette
Xamarin.Android.Support.v7.RecyclerView
Xamarin.Android.Support.Vector.Drawable
Xamarin.Android.Support.VersionedParcelable
Xamarin.Android.Support.ViewPager
Xamarin.Forms.Platform.Android
Xamarin.Forms.Platform
android-crop-1.0.1
```
It looks like AARs and .NET assemblies are both in here, so they'll get unique numbers.